### ISSUE TITLE
Release 0.3.5/0.3.6 causing segfaults with CSV.read

### ISSUE THREAD

**Issue by rofinn** [JuliaData/DataStreams.jl#76]
Looks like the recent releases of DataStreams are causing segfaults while reading a CSV file.

julia 0.6.2
DataStreams 0.3.5 (or 0.3.6)
CSV 0.2.4

```signal (11): Segmentation fault
while loading /root/.julia/v0.6/Foo/test/foo.jl, in expression starting on line 2
unknown function (ip: 0x7f1fd1136b3f)
jl_pchar_to_string at /usr/local/julia/src/array.c:415
string at /root/.julia/v0.6/WeakRefStrings/src/WeakRefStrings.jl:73
wk at /root/.julia/v0.6/WeakRefStrings/src/WeakRefStrings.jl:195
unknown function (ip: 0x7f1f98c0bc16)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_invoke at /usr/local/julia/src/gf.c:41
getindex at /root/.julia/v0.6/WeakRefStrings/src/WeakRefStrings.jl:199
macro expansion at ./broadcast.jl:230 [inlined]
_broadcast! at ./broadcast.jl:219
unknown function (ip: 0x7f1f98c0d0a5)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
broadcast_t at ./broadcast.jl:265
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
jl_apply at /usr/local/julia/src/julia.h:1424 [inlined]
jl_f__apply at /usr/local/julia/src/builtins.c:426
broadcast_c at ./broadcast.jl:321
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
jl_apply at /usr/local/julia/src/julia.h:1424 [inlined]
jl_f__apply at /usr/local/julia/src/builtins.c:426
broadcast at ./broadcast.jl:455
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
jl_apply at /usr/local/julia/src/julia.h:1424 [inlined]
jl_f__apply at /usr/local/julia/src/builtins.c:426
macro expansion at ./util.jl:237 [inlined]
test_client at /root/.julia/v0.6/Foo/test/data.jl:88
test_client at /root/.julia/v0.6/Foo/test/data.jl:62
unknown function (ip: 0x7f1f9adf76f2)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
do_call at /usr/local/julia/src/interpreter.c:75
eval at /usr/local/julia/src/interpreter.c:242
eval_body at /usr/local/julia/src/interpreter.c:543
jl_interpret_toplevel_thunk at /usr/local/julia/src/interpreter.c:692
jl_toplevel_eval_flex at /usr/local/julia/src/toplevel.c:592
jl_parse_eval_all at /usr/local/julia/src/ast.c:873
jl_load at /usr/local/julia/src/toplevel.c:616
include_from_node1 at ./loading.jl:576
unknown function (ip: 0x7f1fb07075c2)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
include at ./sysimg.jl:14
macro expansion at /root/.julia/v0.6/Foo/test/runtests.jl:46 [inlined]
macro expansion at ./test.jl:860 [inlined]
anonymous at ./<missing> (unknown line)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_toplevel_eval_flex at /usr/local/julia/src/toplevel.c:589
jl_parse_eval_all at /usr/local/julia/src/ast.c:873
jl_load at /usr/local/julia/src/toplevel.c:616
include_from_node1 at ./loading.jl:576
unknown function (ip: 0x7f1fcc336aeb)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
include at ./sysimg.jl:14
unknown function (ip: 0x7f1fcc1cb40b)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
process_options at ./client.jl:305
_start at ./client.jl:371
unknown function (ip: 0x7f1fcc344e98)
jl_call_fptr_internal at /usr/local/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /usr/local/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /usr/local/julia/src/gf.c:1926
jl_apply at /usr/local/julia/ui/../src/julia.h:1424 [inlined]
true_main at /usr/local/julia/ui/repl.c:127
main at /usr/local/julia/ui/repl.c:264
__libc_start_main at /lib64/libc.so.6 (unknown line)
unknown function (ip: 0x401595)
Allocations: 115814178 (Pool: 115796879; Big: 17299); GC: 242
```
This is happening when I run `CSV.read(joinpath(p, f))` in my test code, but the file in question loads fine on the REPL.

**Comment by quinnj** [comment]
Could you share the file? Even if privately? Id' like to track this down and reproduce.

**Comment by rofinn** [comment]
Alright, I managed to get a working example with an obfuscated dataset [here](https://github.com/rofinn/DataStreamsTest.jl). I've been testing this by running `Pkg.test("DataStreamsTest")` which will try and load the 35MB git-lfs file `test/test.csv`.

Some things I noticed trying to get this to hit the segfault condition include:

1. Only happens when doing `tuple.(CSV.read("test.csv").columns...)` inside tests (works fine on the REPL).
2. Reducing the size of the dataset by either removing columns or rows avoids the segfault.
